### ISSUE TITLE
Negative exponent with value -1 (minus one) raises error

### ISSUE THREAD

**Issue by edg-stg** [RaRe-Technologies/gensim#3232]
#### Problem description

I try to vary the value of the negative exponent parameter. When I use a value of -1, training works fine, but prediction afterwards with infer_vector does not work and raises the error "ValueError: Integers to negative integer powers are not allowed."

This is due to the following line: https://github.com/RaRe-Technologies/gensim/blob/266a01455ade51a93a08dba5950e87b4d98e0724/gensim/models/word2vec.py#L836

Here, numpy does not raise an integer by the power of another, but negative integer.

I guess this could be solved by converting the exponent to a float in this case?

**Comment by gojomo** [comment]
Yes - I'd expect if `count` (which may have only recently became a `numpy`-typed int rather than a generic Python int) were converted to a float (either Python `float()` or `np.float32()`), this error would be eliminated. 

Can you share the full error stack traceback? I'm curious why this only shows on `.load()`, not earlier – and while the fix is likely just a few characters, ideally we'd want a new minimal unit test that triggers the failure, pre-fix, as well.

**Comment by pranav1610** [comment]
@edg-stg I would love to work on this issue. Is it available to assign?
